### DISCUSSION TITLE
NativeEventEmitter behaviour in new architecture

### DISCUSSION THREAD

**Discussion by intergalacticspacehighway** [reactwg/react-native-new-architecture#74] [Q&A]
Is something changed with NativeEventEmitter on android? Noticed that we pass `null` on android [here](https://github.com/facebook/react-native/blob/main/Libraries/AppState/AppState.js#L54).

Also, do [these methods](https://github.com/facebook/react-native/blob/main/ReactAndroid/src/main/java/com/facebook/react/modules/appstate/AppStateModule.java#L114) trigger on android? NetInfo uses them to [set and reset some flags](https://github.com/react-native-netinfo/react-native-netinfo/blob/master/android/src/main/java/com/reactnativecommunity/netinfo/NetInfoModule.java#L69). These flags are used for optimisation while triggering [event call from native](https://github.com/react-native-netinfo/react-native-netinfo/blob/master/android/src/main/java/com/reactnativecommunity/netinfo/ConnectivityReceiver.java#L121). Thanks 🙏

**Comment by fabOnReact** [comment]
https://github.com/facebook/react-native/pull/34353#issuecomment-1207744977
https://github.com/facebook/react-native/issues/30841#issuecomment-1228128357

I'm also investigating [these changes](https://github.com/fabriziobertoglio1987/react-native/commit/46df827c7ef3baac16d4bf6bcb1a1a90e5ade0ef) in the Java/CPP API which essentially add the surfaceId to the [event](https://github.com/fabriziobertoglio1987/react-native/blob/b80bcb3104c903f97974d90db3827b997af9c7cc/ReactAndroid/src/main/java/com/facebook/react/uimanager/events/Event.java#L60). Is a Fabric optimisation, but does not work with onAccessibililtyAction callback on Fabric


https://github.com/facebook/react-native/blob/28e9c3b87e709dce3ffec208ff5228d07eaad512/ReactAndroid/src/main/java/com/facebook/react/uimanager/ReactAccessibilityDelegate.java#L360-L373

https://github.com/fabriziobertoglio1987/react-native/blob/b80bcb3104c903f97974d90db3827b997af9c7cc/ReactAndroid/src/main/java/com/facebook/react/uimanager/events/EventDispatcherImpl.java#L112

https://github.com/facebook/react-native/blob/163171ccab6937785f4f3c85e011bd14540bebf5/ReactAndroid/src/main/java/com/facebook/react/uimanager/events/ReactEventEmitter.java#L130-L140

https://github.com/fabriziobertoglio1987/react-native/blob/b80bcb3104c903f97974d90db3827b997af9c7cc/ReactCommon/react/renderer/components/view/ViewEventEmitter.cpp#L15

I'm reading this documentation now https://github.com/fabriziobertoglio1987/react-native/blob/b80bcb3104c903f97974d90db3827b997af9c7cc/ReactAndroid/src/main/java/com/facebook/react/uimanager/events/Event.java#L60, but seems to apply to the old architecture

I quote [mrousavy](https://github.com/mrousavy)
>RN Core they always pass null to new NativeEventEmitter(..) on Android. Only on iOS they actually pass the native module. I believe on Android the event names are shared across all event emitters, so the argument is not needed on Android.

https://github.com/facebook/react-native/commit/114be1d2170bae2d29da749c07b45acf931e51e2
https://github.com/facebook/react-native-website/pull/2791
